Handle programs: create and retrieve

KevinTss opened the pull request.

KevinTss committed:
Handle programs: create and retrieve

alvermeylen commented on src/components/program/add-program-modal/AddProgramModal.jsx:
```diff
@@ -0,0 +1,55 @@
+  isOpen,
+  onToggle,
+  createProgram,
+  isCreateProgramLoading,
```
WOuld be better to just user a status, to avoid needing checking and changing several variables to know the state

alvermeylen commented on src/components/program/add-program-modal/AddProgramModal.jsx:
```diff
@@ -0,0 +1,55 @@
+  };
+
+  return (
+    <SideModal isShown={isOpen} onCloseComplete={onToggle}>
```
For later: check if these modals are okay with accessibility requirements.

KevinTss replied:
track [here](https://github.com/coach-marketplace/app/issues/30)

alvermeylen commented on src/components/program/add-program-modal/AddProgramModal.jsx:
```diff
@@ -0,0 +1,55 @@
+const mapStateToProps = (state) => ({
+  isCreateProgramLoading: state.program.actions.create.loading,
+  isCreateProgramSuccess: state.program.actions.create.success,
+  isCreateProgramError: state.program.actions.create.error,
```
As previously discussed, user getters to improve mainteznanbility and readability

alvermeylen commented on src/components/program/program-form/ProgramForm.jsx:
```diff
@@ -0,0 +1,91 @@
+      isPrivate: initialValues.isPrivate || false,
+    },
+    // validationSchema: validationSchema,
+    onSubmit: (values) => onSubmit(values),
```
Can't we make this component self-sufficient by letting it handle the call to create itself ?

KevinTss replied:
Be careful in that case because we should be able to use this form for a creation but also for an updating. So IMO it should be the parent to take care of the action

alvermeylen commented on src/components/program/program-form/validation.js:
```diff
@@ -0,0 +1,10 @@
+import * as Yup from "yup";
+
+export default Yup.object({
+  email: Yup.string()
```
add .matches(/^.*(@).*(\.).*$/m, "invalid format for email address") to ensure email address is in the right format

KevinTss replied:
We can use directly the method from [Yup](https://github.com/jquense/yup#stringemailmessage-string--function-schema) instead

alvermeylen commented on src/components/program/programs-container/ProgramsContainer.jsx:
```diff
@@ -0,0 +1,62 @@
+import { retrieveAll as retrieveAllPrograms } from "../../../store/modules/program/actions";
+
+const ProgramsContainer = ({
+  isFetchProgramsLoading,
+  isFetchProgramsSuccess,
```
one status is clearer

alvermeylen commented on src/store/modules/program/actions.js:
```diff
@@ -0,0 +1,64 @@
+        dispatch(createSuccess(response.data));
+      })
+      .catch((error) => {
+        dispatch(createFailed(error.message));
```
error.message does not exist. you should use error.response.data.public_message

alvermeylen commented on src/store/modules/program/state.js:
```diff
@@ -0,0 +1,9 @@
+
+export default {
+  list: [],
+  actions: {
```
to be discussed: why do we use the action entry in the state ? Does that make sense?

KevinTss committed:
update the way to add, edit and delete an exercise

KevinTss committed:
Merge branch 'staging' into programs

KevinTss committed:
imp: improve design

KevinTss committed:
improve CRUD exercise and UI

- Change & fix the way to update an exercise
- Improve UI for workout form using the Editor input

KevinTss committed:
Add google on auth form

KevinTss committed:
fix editor doesn't accept default value

KevinTss committed:
Improve CRUD workout

KevinTss committed:
improve design

KevinTss committed:
fix: button where reachable twice with keyboard

Due to the combinaison of Button inside a A, when we navigate from keyboard, we had 2 times a focus on the "same" element for the user perspective
That's why we change the tagname of the Button to a div

KevinTss committed:
refactor routes

Make them depending of an array

KevinTss committed:
fix some design consistencies

KevinTss committed:
imp: main is now usable with a full layout

- Also remove useless code
- Change design of layout spacing

KevinTss committed:
imp(UI): Improve UI adding dialog

- Add dialog modal
- Move Text into typography folder
- Refactor all components previously use import to direct UI component and make them import from the global UI file

KevinTss committed:
Add fetch single workout node in store

KevinTss committed:
Add Workout and Program class (not use yet)

KevinTss committed:
Add Update and delete program in node store

KevinTss committed:
fix: add forgoten UI component

KevinTss committed:
Imp: add program edition page

A coach can access an edition program page where he can find the program add he's able to add a workout to a specific day of the program

KevinTss committed:
imp: make able the coach to re-order workouts

KevinTss committed:
fix crud exercises and workouts

KevinTss committed:
Improve design

KevinTss committed:
fix: adding an innexisting user was failing
